crates/rust-analyzer/src/config.rs

```diff
@@ -196,110 +175,113 @@ config_data! {
         /// and a blue icon in the `Problems Panel`.
         diagnostics_warningsAsInfo: Vec<String> = "[]",
 
-        /// Expand attribute macros.
-        experimental_procAttrMacros: bool = "true",
-
-        /// Controls file watching implementation.
-        files_watcher: String = "\"client\"",
         /// These directories will be ignored by rust-analyzer. They are
         /// relative to the workspace root, and globs are not supported. You may
         /// also need to add the folders to Code's `files.watcherExclude`.
         files_excludeDirs: Vec<PathBuf> = "[]",
+        /// Controls file watching implementation.
+        files_watcher: String = "\"client\"",
 
-        /// Enables highlighting of related references while hovering your mouse above any identifier.
-        highlightRelated_references: bool = "true",
-        /// Enables highlighting of all exit points while hovering your mouse above any `return`, `?`, or return type arrow (`->`).
-        highlightRelated_exitPoints: bool = "true",
         /// Enables highlighting of related references while hovering your mouse `break`, `loop`, `while`, or `for` keywords.
-        highlightRelated_breakPoints: bool = "true",
+        highlightRelated_breakPoints_enable: bool = "true",
+        /// Enables highlighting of all exit points while hovering your mouse above any `return`, `?`, or return type arrow (`->`).
+        highlightRelated_exitPoints_enable: bool = "true",
+        /// Enables highlighting of related references while hovering your mouse above any identifier.
+        highlightRelated_references_enable: bool = "true",
         /// Enables highlighting of all break points for a loop or block context while hovering your mouse above any `async` or `await` keywords.
-        highlightRelated_yieldPoints: bool = "true",
-
-        /// Use semantic tokens for strings.
-        ///
-        /// In some editors (e.g. vscode) semantic tokens override other highlighting grammars.
-        /// By disabling semantic tokens for strings, other grammars can be used to highlight
-        /// their contents.
-        highlighting_strings: bool = "true",
-
-        /// Whether to show documentation on hover.
-        hover_documentation: bool       = "true",
-        /// Use markdown syntax for links in hover.
-        hover_linksInHover |
-        hoverActions_linksInHover: bool = "true",
+        highlightRelated_yieldPoints_enable: bool = "true",
 
         /// Whether to show `Debug` action. Only applies when
         /// `#rust-analyzer.hoverActions.enable#` is set.
-        hoverActions_debug: bool           = "true",
+        hover_actions_debug_enable: bool           = "true",
         /// Whether to show HoverActions in Rust files.
-        hoverActions_enable: bool          = "true",
+        hover_actions_enable: bool          = "true",
```

Comment:
Right, so some of these also have global enable toggles. I am unsure whether it makes sense to have them, part of me says part of me says no. Basically all they really allow is to easily turn everything off, but if one wants all enabled they have to have this enabled AND each thing separately as well